### PR TITLE
Fix #1945: Optimise the onboarding_slide.xml file

### LINKED ISSUE #1945: Optimise code in onboarding_slide.xml
This issue is tracking the optimisation work of `onboarding_slide.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all textviews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

While making the above changes the UI should not get affected and the end result should still be same as before (or an optimised version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by Austinepass**
Fix #1945 Style the TextViews by using styles in the styles.xml file

### Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's not called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is assigned to an appropriate reviewer in both the Assignees and the Reviewers sections.

### CODE REVIEW

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
       <TextView
         android:id="@+id/slide_title_text_view"
+        style="@style/Heading1"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml]:
```
android:layout_width="0dp"
android:layout_height="wrap_content"
```

The above lines can be removed and you can introduce `horizontalBias`.

Considering that the Heading1 already have these attributes.

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 2]
```diff
@@ -46,24 +47,22 @@
         android:gravity="center_horizontal"
         android:text="@{viewModel.title}"
         android:textColor="@color/oppiaPrimaryText"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml]: I think this color is part of `Heading1`, if so can you remove this attribute here?

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 3]
```diff
@@ -46,24 +47,22 @@
...
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintHorizontal_bias="1.0"
         app:layout_constraintStart_toEndOf="@+id/slide_image_view"
         app:layout_constraintTop_toTopOf="parent" />
 
       <TextView
         android:id="@+id/slide_description_text_view"
+        style="@style/Subtitle1"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml]:
```
android:layout_width="0dp"
android:layout_height="wrap_content"
```

The above lines can be removed and you can introduce `horizontalBias`.

Considering that the Subtitle1 already have these attributes.

File: app/src/main/res/layout-land/onboarding_slide.xml [hunk 4]
```diff
@@ -46,24 +47,22 @@
...
         android:layout_marginStart="20dp"
         android:layout_marginTop="16dp"
         android:layout_marginEnd="40dp"
-        android:fontFamily="sans-serif"
         android:gravity="center_horizontal"
         android:text="@{viewModel.description}"
         android:textColor="@color/oppiaPrimaryText"
```
Comment (rt4914) [inline app/src/main/res/layout-land/onboarding_slide.xml]: I think this color is part of Subtitle1, if so can you remove this attribute here?

**Review by rt4914 — CHANGES_REQUESTED** [review]
Thanks @Austinepass 

I have suggested changes in 1 file and can you apply those changes in all files?

Thanks.

**Comment by rt4914** [comment]
Also, please add the `Checklist` back to the description, which you can find here. https://github.com/oppia/oppia-android/pull/1960#issue-500151073

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 1 above)
Comment (Austinepass) [reply app/src/main/res/layout-land/onboarding_slide.xml:42]: @rt4914 The Heading1 style has basically two attributes, the  fontFamily and the textSize, can you take another look at it.
  <style name="Heading1" parent="TextView.Common1">
    <item name="android:fontFamily">sans-serif</item>
    <item name="android:textSize">24sp</item>
    <item name="fontFamily">sans-serif</item>
  </style>

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 4 above)
Comment (Austinepass) [reply app/src/main/res/layout-land/onboarding_slide.xml]: Done.

**Comment by Austinepass** [comment]
@rt4914 I have effected the requested changes, kindly review. Thanks.

**Comment by rt4914** [comment]
@Austinepass Couple of points to keep in mind:

1. Reply to each and every comment.
2. The person who has started the comment thread should resolve the comment thread. So can unresolve the two comments which you resolved.

Thanks.

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 1 above)
Comment (Austinepass) [reply app/src/main/res/layout-land/onboarding_slide.xml]: I have removed the redundant `android:layout_height="wrap_content"` line.

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 2 above)
Comment (Austinepass) [reply app/src/main/res/layout-land/onboarding_slide.xml]: Yes, I have now removed the ` android:textColor="@color/oppiaPrimaryText"` in the latest commit

File: app/src/main/res/layout-land/onboarding_slide.xml (hunk 3 above)
Comment (Austinepass) [reply app/src/main/res/layout-land/onboarding_slide.xml]: I have also implemented the suggested changes.

**Comment by Austinepass** [comment]
> @Austinepass Couple of points to keep in mind:
> 
> 1. Reply to each and every comment.
> 2. The person who has started the comment thread should resolve the comment thread. So can unresolve the two comments which you resolved.
> 
> Thanks.

@rt4914 Noted with thanks. I have now unresolved the two comments and replied to them.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**